test/common/http/http1/conn_pool_test.cc

```diff
@@ -128,9 +129,12 @@ class Http1ConnPoolImplTest : public testing::Test {
         conn_pool_(dispatcher_, cluster_, upstream_ready_timer_) {}
 
   ~Http1ConnPoolImplTest() {
-    // Make sure all gauges are 0.
+    // Make sure all gauges are 0, except the circuit_breaker remaining resource
+    // gauges which default to the resource max.
     for (const Stats::GaugeSharedPtr& gauge : cluster_->stats_store_.gauges()) {
-      EXPECT_EQ(0U, gauge->value());
+      if (!absl::StrContains(gauge->name(), "remaining")) {
```

Comment:
This seems a little fragile to me. Is it possible to do a regex match here which matches on "circuit_breaker.*.remaining"? Same below? Also given this pattern has been copy/pasted into a bunch of different tests, maybe move into a utility function in case we need to do anything like this in the future?

Reply:
Sure thing, I gave this a shot. Thanks for taking a look.